examples/index.html

```diff
@@ -26,28 +36,49 @@
         setRequestHeader(name, value) {
           // Do nothing.
         }
-        getResponseHeader(name, value) {
-          // Do nothing.
+        getResponseHeader(name) {
+          if (name === 'Content-Type' && focusedInput.classList.contains('json')) {
+            return 'application/json'
+          }
         }
         send() {
-          this.status = input.value === '422' ? 422 : 200
+          if (focusedInput.classList.contains('json')) {
+            this.responseText = '{"text": "success", "html": "<b>success</b>"}'
+          } else {
+            this.responseText = 'success'
+          }
+          this.status = focusedInput.value === '422' ? 422 : 200
           setTimeout(this.onload.bind(this), 0)
         }
       }
       window.XMLHttpRequest = FakeXMLHttpRequest
-      const state = document.getElementById('state')
-      document.addEventListener('auto-check-send', () => {
-        state.textContent = 'loading'
-        button.disabled = true
-      })
-      document.addEventListener('auto-check-success', () => {
-        state.textContent = 'success'
-        button.disabled = false
-      })
-      document.addEventListener('auto-check-error', () => {
-        state.textContent = 'error'
-        button.disabled = true
-      })
+
+      for (const form of document.querySelectorAll('form')) {
+        const formInput = form.querySelector('input')
+        const button = form.querySelector('button')
+        const state = form.querySelector('.state')
+
+        formInput.addEventListener('focus', () => {
+          focusedInput = formInput
+        })
+
+        form.addEventListener('auto-check-send', () => {
+          state.textContent = 'loading'
+          button.disabled = true
+        })
+        form.addEventListener('auto-check-success', event => {
+          if (focusedInput.classList.contains('json')) {
+            state.innerHTML = event.detail.message.html
```

Comment:
<img width="300" alt="image" src="https://user-images.githubusercontent.com/1153134/61911859-b0a86000-af06-11e9-9373-d095475eb757.png">

```suggestion
            state.innerHTML = JSON.parse(event.detail.message).html
```